smac_planner/src/a_star.cpp

```diff
@@ -437,11 +456,6 @@ float AStarAlgorithm<NodeT>::getHeuristicCost(const NodePtr & node)
   float heuristic = NodeT::getHeuristicCost(
     node_coords, _goal_coordinates) * _neutral_cost;
 
-  // If we're far from goal, we want to ensure we can speed it along
-  if (heuristic > getToleranceHeuristic()) {
```

Comment:
This was a very helpful optimization for the 2D A* search. I understand that it broke some of the rules for a good heuristic, but I failed to find a situation where it didn't help expand more quickly towards the goal, and then slows down to be admissible once in a tolerance value.

Reply:
Offending commit removed.